### PR TITLE
Inorganic mobs no longer have weird, undocumented resistances

### PR DESCRIPTION
**Pull request by RikuTheKiller**
## About The Pull Request

I found out about this stuff during a powergaming test of probital and modafinil.
What ensued was me being in infsleep, so I dug deeper.

As it turns out, all skeletons, golems, plasmamen, etc. have immunity to reagent toxin damage and reagent stamina damage.
This PR rectifies that as these are things that simply aren't documented anywhere and don't even make sense gameplay-wise since all of these races are affected by the other effects of these reagents. (including burn/brute damage)

Also fixed some formatting stuff, force checks using 0 instead of FALSE and added support for a few biotype things in some of the damage procs along the way.

Oh yeah and this also means plasmamen are affected by toxin damage in general now. Same goes for others. This MAY cause issues with stuff like plasma fixation, although it really shouldn't.
## Why It's Good For The Game

Undocumented, likely unintentional mechanics that cause balancing issues simply shouldn't stay around.

And neither should shitcode.
## Changelog
:cl:
fix: Removed undocumented toxin and stamina damage resistances from all inorganic species except androids.
refactor: Properly implemented some checks for a few damage procs, they were unused so this shouldn't affect anything.
/:cl:

### CODE REVIEW

**Comment by Helg2** [comment]
I think it was mentioned somewhere that plasmamen are immune to most of toxins or something, which makes sense as they are fungus on the bones.
Skeletons are just bones, and golems are piles of rock, I don't see why would they care about toxin damage if they aren't organic.

File: code/modules/mob/living/carbon/damage_procs.dm [hunk 1]
```diff
@@ -87,8 +87,8 @@
 		return
 	adjustFireLoss(diff, updating_health, forced, required_bodytype)
 
-/mob/living/carbon/adjustToxLoss(amount, updating_health = TRUE, forced = FALSE, required_biotype = MOB_ORGANIC)
-	if(!forced && !(mob_biotypes & required_biotype))
+/mob/living/carbon/adjustToxLoss(amount, updating_health = TRUE, forced = FALSE, required_biotype)
+	if(!forced && required_biotype && !(mob_biotypes & required_biotype))
```
[the remark this answers is not in the document]
Comment (RikuTheKiller) [reply code/modules/mob/living/carbon/damage_procs.dm:91]: false, tested with different mob_biotypes and such, this is simply incorrect

it checks if the flag is set at all and if it is set, then it checks if your biotype doesnt match

**Comment by the-orange-cow** [comment]
To my understanding, skeletons aren't able to metabolise most medicines.  If they become vulnerable to toxin damage, is it possible for them to heal it?

File: code/modules/mob/living/carbon/damage_procs.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (ZephyrTFA) [reply code/modules/mob/living/carbon/damage_procs.dm:91]: @vinylspiders (1<<0) is 1 not 0

**Comment by RikuTheKiller** [comment]
> To my understanding, skeletons aren't able to metabolise most medicines. If they become vulnerable to toxin damage, is it possible for them to heal it?

They can metabolize all medicine, however any toxin or stamina damage related effects won't apply to them.
Yeah... it's weird.

File: code/modules/mob/living/carbon/damage_procs.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (RikuTheKiller) [reply code/modules/mob/living/carbon/damage_procs.dm:91]: If I'm going to change it to use this, I'd rather rename the argument to "affected_biotypes" as that makes way more sense with no flags meaning it affects nothing and ALL meaning it affects all biotypes.

**Comment by Likteer** [comment]
@RikuTheKiller My good Sir, you just made a balance PR. Those resistances are intended. https://github.com/tgstation/tgstation/pull/72664

**Comment by Fikou** [comment]
@RikuTheKiller dont troll with labels

**Comment by Pickle-Coding** [comment]
The documentation is from the fact that they're skeletons, and one of them is literally surrounded by toxins in order to live.

**Comment by RikuTheKiller** [comment]
> The documentation is from the fact that they're skeletons, and one of them is literally surrounded by toxins in order to live.

They're surrounded by plasma. The documentation is such that they're immune to plasma, which is stated in their species entry.
Nothing about total toxin immunity is stated anywhere.
Also we should probably address the issue on stamloss. It results in stuff like probital infsleep and drugs giving their ordinary effects but leaving out the stamloss related ones.

**Pull request closed without merging**